client_perms/server.sh

```diff
@@ -0,0 +1,278 @@
+
+	gluster volume start ${GLUSTER_VOLUME} force
+
+	#disable gluster-nfs
+	#gluster v set vol1 nfs.disable on
+	#sleep 2
+
+	#enable cache invalidation
+	#gluster v set vol1 cache-invalidation on
+
+	# TODO: open only the ports needed?
+	# disable the firewall, otherwise the client can not connect
+	systemctl stop firewalld || service iptables stop
+
+	# TODO: SELinux prevents creating special files on Gluster bricks (bz#1331561)
+	setenforce 0
+
+	# Export the volume
+	mkdir -p /usr/libexec/ganesha
+	cd /usr/libexec/ganesha
+	yum -y install wget
+	wget https://raw.githubusercontent.com/gluster/glusterfs/release-3.10/extras/ganesha/scripts/create-export-ganesha.sh
+	wget https://raw.githubusercontent.com/gluster/glusterfs/release-3.10/extras/ganesha/scripts/dbus-send.sh
+	chmod 755 create-export-ganesha.sh dbus-send.sh
+
+	/usr/libexec/ganesha/create-export-ganesha.sh /etc/ganesha on ${GLUSTER_VOLUME}
+	/usr/libexec/ganesha/dbus-send.sh /etc/ganesha on ${GLUSTER_VOLUME}
+
+	# wait till server comes out of grace period
+	sleep 90
+
+	conf_file="/etc/ganesha/exports/export."${GLUSTER_VOLUME}".conf"
+
+	#Parsing export id from volume export conf file
+	export_id=$(grep 'Export_Id' ${conf_file} | sed 's/^[[:space:]]*Export_Id.*=[[:space:]]*\([0-9]*\).*/\1/')
+
+	# basic check if the export is available, some debugging if not
+	if ! showmount -e | grep -q -w -e "${GLUSTER_VOLUME}"
+	then
+		echo "+++ /var/log/ganesha.log +++"
+		cat /var/log/ganesha.log
+		echo
+		echo "+++ /etc/ganesha/ganesha.conf +++"
+		grep --with-filename -e '' /etc/ganesha/ganesha.conf
+		echo
+		echo "+++ /etc/ganesha/exports/*.conf +++"
+		grep --with-filename -e '' /etc/ganesha/exports/*.conf
+		echo
+		echo "Export ${GLUSTER_VOLUME} is not available"
+		exit 1
+	fi
+
+	echo "============Displaying Initial Configuration File============================="
+	cat /etc/ganesha/exports/export.${GLUSTER_VOLUME}.conf
+
+	#Enabling ACL for the volume if ENABLE_ACL param is set to True
+	if [ "${ENABLE_ACL}" == "True" ]
+	then
+	  sed -i s/'Disable_ACL = .*'/'Disable_ACL = false;'/g ${conf_file}
+	  cat ${conf_file}
+
+	  dbus-send --type=method_call --print-reply --system  --dest=org.ganesha.nfsd /org/ganesha/nfsd/ExportMgr  org.ganesha.nfsd.exportmgr.UpdateExport string:/etc/ganesha/exports/export.${GLUSTER_VOLUME}.conf string:"EXPORT(Export_Id = 2)"
+	fi
+
+fi
+
+if [ "$1" = "server_stage1" ]
+then
+	echo "======= SERVER STAGE 1 =========="
+	touch clientBlock.txt
+	echo -e "CLIENT{
+	\tClients = ${CLIENT};
+        \tSquash = \"no_root_squash\";
+        \tAccess_type = \"RO\";
+        \tProtocols = \"3\",\"4\";
+	}" > clientBlock.txt
+	
+	conf_file="/etc/ganesha/exports/export."${GLUSTER_VOLUME}".conf"
+
+	#Parsing export id from volume export conf file
+	export_id=$(grep 'Export_Id' ${conf_file} | sed 's/^[[:space:]]*Export_Id.*=[[:space:]]*\([0-9]*\).*/\1/')
+
+	echo "=======CLIENTBLOCK========="
+	cat clientBlock.txt
+	sed -i '19e cat clientBlock.txt' ${conf_file}
```

Comment:
This might fail if the default config file generated by create-ganesha-export.sh changes.  It might be better to look for a certain line in the file and append after it.

Reply:
@dang ... Yea .. got it .. :)